sql/src/main/java/io/crate/operation/operator/any/AnyEqOperator.java

```diff
@@ -30,14 +30,17 @@
 import io.crate.sql.tree.ComparisonExpression;
 import io.crate.types.DataType;
 import io.crate.types.DataTypes;
-import io.crate.types.SetType;
+import io.crate.types.SingleColumnTableType;
 
 public class AnyEqOperator extends AnyOperator {
 
     public static final String NAME = OPERATOR_PREFIX + ComparisonExpression.Type.EQUAL.getValue();
 
     public static FunctionInfo createInfo(DataType targetType) {
-        return new FunctionInfo(new FunctionIdent(NAME, ImmutableList.of(targetType, new SetType(targetType))), DataTypes.BOOLEAN);
+        return new FunctionInfo(
+            new FunctionIdent(NAME,
+                ImmutableList.of(targetType, new SingleColumnTableType(targetType))),
```

Comment:
discussed this already with @matriv 

this looks like this function only exists for singlecolumntabletype? is that right, what about 1 = any([1,2])? 

either we have here a public static method that is called only for one case of others, which is misleading - or we have no support for array types in the any operators and cast it?